smart-contracts/contracts/airdrop/src/admin.rs

```diff
@@ -0,0 +1,365 @@
+/// in the event for each removed user.
+pub fn execute_remove_users(deps: DepsMut, users: Vec<String>) -> Result<Response, AirdropErrors> {
+    // Load the current airdrop configuration from storage
+    let current_airdrop_config = AIRDROP_CONFIG.load(deps.storage)?;
+
+    // Check if the current airdrop window is not open (start_height or end_height not zero)
+    if current_airdrop_config.start_height != 0 || current_airdrop_config.end_height != 0 {
+        return Err(AirdropErrors::InvalidChangeUserInfo {});
+    }
+
+    // Initialize vectors to store removed users and attributes for the event
+    let mut removed_users: Vec<String> = Vec::new();
+    let mut attributes: Vec<Attribute> = Vec::new();
+
+    // Iterate through the list of users to be removed
+    for user in users.iter() {
+        // Validate the user's address
+        deps.api.addr_validate(user)?;
+
+        // Load the user_info entry from storage
+        let user_info = USER_INFO.load(deps.storage, user.to_string())?;
+
+        // Check if the claimed flag is false, indicating that the user has not claimed
+        if !user_info.get_claimed_flag() {
+            // Add the user's address to the list of removed users
+            removed_users.push(user.to_string());
+
+            // Remove the user's entry from the USER_INFO map
+            USER_INFO.remove(deps.storage, user.to_string());
+
+            // Add user address as an attribute for the event
+            attributes.push(Attribute {
+                key: "address".to_string(),
+                value: user.to_string(),
+            });
+        }
+    }
+
+    // Return a default response if all checks pass with an "airdrop_remove_users" event
+    Ok(
+        Response::default()
+            .add_event(Event::new("airdrop_remove_users").add_attributes(attributes)),
+    )
+}
+
+/// Withdraws airdrop funds to the specified address after the airdrop window has ended.
+///
+/// # Arguments
+///
+/// * `deps` - Dependencies to access storage, external data, and assets.
+/// * `env` - Environment information including the current block height.
+/// * `withdraw_address` - The address to which the airdrop funds will be withdrawn.
+///
+/// # Errors
+///
+/// Returns an error if the current block height is not within the airdrop window or the window is open-ended.
+/// Also returns an error if the withdrawal address is invalid.
+///
+/// # Returns
+///
+/// Returns a response indicating the success of the withdrawal and includes attributes in the response for tracking.
+pub fn execute_withdraw_funds(
+    deps: DepsMut,
+    env: Env,
+    withdraw_address: String,
+) -> Result<Response, AirdropErrors> {
+    // Load the current airdrop configuration from storage
+    let current_airdrop_config = AIRDROP_CONFIG.load(deps.storage)?;
+
+    // Check if the current block height is within the airdrop window or the window is open-ended
+    if env.block.height < current_airdrop_config.end_height
```

Comment:
could be divided in to two checks,
1. is_window_initialised(..) ->  ( current_airdrop_config.end_height == || current_airdrop_config.start_height == 0 )
2. is_airdrop_ended(..) ->  ( env.block.height 
3.  > current_airdrop_config.end_height )

Reply:
@arhamchordia For future reference, please integrate this, makes reading code easier

Reply:
Sounds good.